plugins/builds/steps/active.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const getSchema = schema.models.build.getStep;
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/builds/{id}/steps/active',
+    config: {
+        description: 'Get a step for a build',
+        notes: 'Returns a step record',
+        tags: ['api', 'builds', 'steps'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', 'build', 'pipeline', '!guest', 'temporal']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const buildFactory = request.server.app.buildFactory;
+            const buildIdCred = request.auth.credentials.username
+                && request.auth.credentials.username.toString();
+            const buildId = request.params.id && request.params.id.toString();
+
+            if (request.auth.credentials.scope.includes('temporal') && buildId !== buildIdCred) {
+                return reply(boom.forbidden(`Credential only valid for ${buildIdCred}`));
+            }
+
+            return buildFactory.get(buildId)
+                .then((buildModel) => {
+                    if (!buildModel) {
+                        throw boom.notFound('Build does not exist');
+                    }
+                    const stepModel = buildModel.steps.find(
+                        step => step.startTime && !step.endTime);
+
+                    if (!stepModel) {
+                        throw boom.notFound('Active step does not exist');
```

Comment:
should it throw error or just return empty?

Reply:
Well, I think both works, returns a 404, and in worker we are looking for status 200.

Reply:
I think this endpoint should return empty if it cannot find anything